### PR TITLE
R4R: BEP84 mirror and sync

### PR DESCRIPTION
**Pull request by HaoyangLiu**
Currently, if a user wants to issue and bind a BEP2 token with an existing BEP20 token, it has to do a set of complex operations, including issue BEP2 tokens, sending a bind transaction and approving the binding request. The BEP will bring a mechanism to simplify the above process. In the new mechanism, what anyone can do this is just sending a transaction to Binance Smart Chain.

### CODE REVIEW

**Comment by BNB48ClubSirIan** [comment]
This can be excellent!
Actually, we are currently considering issuing the BEP2 version of our token and bind them, but the protocol is too tricky for us even on test net.

Looking forward to this BEP

File: BEP84.md [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+## 3. Motivation
+Current bind mechanism is based on the context that our community members are on the Binance Chain and they want to extend their tokens to the Binance Smart Chain. However, with the evolution of our community, things changed. In most cases, users issue bep20 on the Binance Smart Chain first without considering whether they will issue BEP2 on the Binance Chain or not,  so a new mechanism to conveniently extend BEP20 assets to the Binance Chain is required. In addition, the new mechanism will encourage users to extend their assets to the Binance Chain which is very helpful to flourish the Binance Chain community. 
+
+## 4. Status
+This BEP is under draft.
+ 
+## 5. Specification
+
+Two new permissionless methods will be imported into TokenManager contract:
+Mirror: If a BEP20 contract is not bound to any BEP2 token, anyone can call the mirror method to automatically issue a BEP2 token and bind them. The user is not required to have any BEP20 token and just needs to pay enough BEP2 issue fee. Besides, the BEP20 contract is even not required to implement getOwner method. After binding, all circulation is on BSC.
```
Comment (guagualvcha) [inline BEP84.md:41]: `all circulation is on BSC` => `all the initial circulation is on BSC`

File: BEP84.md [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+| **Param Name**    | **Type** | **Description**   |
+| ------------ | -------- | ---------------------- |
+| BEP20Addr    | Address  | BEP20 contract address |
+| ExpiredTime  | uint64   | The deadline to deliver this package on BC |
+| msg.value    | uint256  | Sum of cross chain fee and mirror fee |
+
+##### 5.1.1.2 Pre-check
+
+1. Ensure the BEP20 token hasn’t been bound before and is not in mirror pending status.
+2. Ensure the BEP20 symbol follows the BEP2 symbol requirements.
```
Comment (guagualvcha) [inline BEP84.md:59]: \+Ensure the BEP20 name follows the BEP2 name requirements.

File: BEP84.md [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+3. Ensure the equivalent total supply is no greater than Max BEP2 total supply.
+4. Ensure msg.value >= MirrorFee + CrossChainFee and msg.value = N * 10^10
+5. Ensure expired time is earlier than one day later and no earlier than two minutes later.
+
+##### 5.1.1.3 Core Mechanism
+
+1. Transfer relayFee to TokenHub
```
Comment (guagualvcha) [inline BEP84.md:66]: Above you talk about `CrossChainFee`, but it is the same as `relayFee` here,   please use a same name.

File: BEP84.md [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+2. Mark the BEP20 token is in mirror pending status
```
Comment (guagualvcha) [inline BEP84.md:67]: `Mark the BEP20 token is in mirror pending status` ==> `Mark the BEP20 token as mirror pending status`

File: BEP84.md [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+| BEP20Addr         | Address  | BEP20 token address    |
+| BEP20Name         | bytes32  | BEP20 token name       |
+| BEP20Symbol       | bytes32  | BEP20 token symbol     |
+| BEP20Supply       | uint256  | BEP20 total supply     |
+| BEP20Decimals     | uint8    | BEP20 decimals         |
+| MirrorFee         | uint256  | The mirror fee from users, which will be used to cover issue BEP2 token. |
+| ExpiredTime       | uint64   | The package expired time, counted by second |
+
+4. Call CrossChain contract to send a cross chain package.
+5. If a BEP20 token is in mirror pending status, reject approveBind on it
```
Comment (guagualvcha) [inline BEP84.md:82]: or already mirrored?

File: BEP84.md [hunk 6]
```diff
@@ -0,0 +1,216 @@
...
+| **Param Name**    | **Type** | **Description**        |
+| ----------------- | -------- | ---------------------- |
+| BEP20Addr         | Address  | BEP20 contract address |
+| ExpiredTime       | uint64   | The deadline to deliver this package on BC |
+| msg.value         | uint256  | Sum of cross chain fee and sync fee. |
+
+##### 5.1.2.2 Pre-check
+
+1. Ensure the BEP20 token has already been bound.
+2. Ensure the BEP20 is bound by mirror.
```
Comment (guagualvcha) [inline BEP84.md:124]: Step 1 may be not neccesary?

File: BEP84.md [hunk 7]
```diff
@@ -0,0 +1,216 @@
...
+3. Ensure msg.value >= SyncFee + CrossChainFee and msg.value = N * 10^10
+4. Ensure the equivalent total supply on BC doesn’t exceed the maximum limit
+5. Ensure expired time is earlier than one day later and no earlier than two minutes later.
+
+##### 5.1.2.3 Core mechanism
+
+1. Transfer relayFee to TokenHub
```
Comment (guagualvcha) [inline BEP84.md:131]: So do here. "CrossChainFee" and  "relayFee" are the same.

**Review by guagualvcha — APPROVED** [review]

**Review by guagualvcha — APPROVED** [review]

**Review by yutianwu — APPROVED** [review]

**Pull request merged**